test/jasmine/tests/calcdata_test.js

```diff
@@ -28,4 +29,215 @@ describe('calculated data and points', function() {
             expect(gd.calcdata[0][3]).toEqual({ x: false, y: false});
         });
     });
+
+    xdescribe('category ordering', function() {
```

Comment:
@etpinard Thanks :-) Yes, I guess we'll have at least double of the number of test cases (corner cases etc.). Just didn't want to race ahead of your initial feedback and instead, I looked into the unrelated bug you sent my way. Thanks for these specific suggestions.